nomenclate/core/nomenclature.py

```diff
@@ -152,81 +301,81 @@ def get(self, **kwargs):
             start (int): optional definition of start position
         Returns (list): generated object names
         """
-        var_orig = self.var.get()
+        # TODO: rework this entire function, very dirty function.
+        var_attr = self.token_dict.get_token_attr(self.VARIATION_PATH[-1])
+        var_orig = var_attr.get()
+
+        var_start, n_type = self._get_alphanumeric_index(var_orig)
 
-        var_start, n_type = self._get_alphanumeric_index(self.var.get())
-        # Just in case the start hasn't been overridden it's based on the current var_opt index
         if start is None:
             start = var_start
         names = []
-        for index in range(start, end+1):
+        for index in range(start, end + 1):
             if n_type in ['char_hi', 'char_lo']:
                 capital = True if n_type == 'char_hi' else False
-                self.var.set(self.get_variation_id(index, capital))
+                var_attr.set(self.get_variation_id(index, capital))
 
             else:
-                self.var.set(str(index))
+                var_attr.set(str(index))
             if 'var' in kwargs:
                 kwargs.pop("var", None)
             names.append(self.get(**kwargs))
-        self.var.set(var_orig)
+        var_attr.set(var_orig)
         return names
 
+    def update_token_attributes(self):
+        instance_token_attributes = [value for attr, value in iteritems(self.__dict__) if isinstance(value, TokenAttr)]
+        current_token_attributes = self.token_dict.get_token_attrs()
 
-    def get_token_attrs(self):
-        return [token for token in self.__dict__ if isinstance(token, NameAttr)]
+        for token_attribute in current_token_attributes:
+            setattr(self, token_attribute.token, token_attribute)
+            if token_attribute in instance_token_attributes:
+                instance_token_attributes.remove(token_attribute)
 
-    def get_state(self, input_dict=None):
-        """ Returns the current state of dictionary items with hashes to resolve conflicts
+        for excess_token_attr in instance_token_attributes:
+            delattr(self, excess_token_attr.token)
+
+    def _compose_name(self):
+        pass
+
+    def _validate_format_string(self, format_target):
+        """ Checks to see if the target format string follows the proper style
         """
-        if input_dict is None:
-            input_dict = self.__dict__
-        if input_dict:
-            input_dict = self.get_token_attrs()
-            return {item: (input_dict[item].get(), item.__hash__()) for (item, value) in input_dict}
-        return None
+        format_order = self.get_format_order_from_format_string(format_target)
+        separators = '\\._-?'
+
+        format_target = format_target.lower()
+
+        for format_str in format_order:
+            format_target = format_target.replace(format_str.lower(), '')
+        for char in format_target:
+            if char not in separators:
```

Comment:
just to be sure, every character in `format_target` should be one of `\`, `.`, `_`, `-`, or `?`, and nothing else?